pkg/apis/service/types.go

```diff
@@ -26,10 +26,51 @@ type DNSConfig struct {
 
 	// DNSProviderReplication contains enablement for replication of DNSProviders from shoot cluster to control plane
 	DNSProviderReplication *DNSProviderReplication
+
+	// Providers is a list of additional DNS providers that shall be enabled for this shoot cluster.
+	// The primary ("external") provider at `spec.dns.provider` is added automatically
+	// +optional
+	Providers []DNSProvider
+
+	// SyncProvidersFromShootSpecDNS is an optional flag for migrating and synchronising the providers given in the
+	// shoot manifest at section `spec.dns.providers`. If true, any direct changes on the `providers` section
+	// are overwritten with the content of section `spec.dns.providers`.
+	// +optional
+	SyncProvidersFromShootSpecDNS *bool
```

Comment:
Why is this configurable here for end-users? Shouldn't this be the default?

Reply:
Yes, this is the default. The idea here was to allow the end user to disable it in case they want to use the providerConfig as the source of truth.
Should I rename it to `DontSyncProvidersFromShootSpecDNS`? An alternative is to drop it. Then the section is synched from the `spec.dns.provider` until switching to apiVersion `core.gardener.cloud/v1`.

Reply:
I understand, thanks. Let's keep it as you suggested. Once this is out we should ask users to migrate from `.spec.dns.providers` to the `providerConfig`, and after some time we then add validation for preventing any further usage of `.spec.dns.providers`.
Thanks!